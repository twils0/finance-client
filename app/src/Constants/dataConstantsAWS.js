// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
