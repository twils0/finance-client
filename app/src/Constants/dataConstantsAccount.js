export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
