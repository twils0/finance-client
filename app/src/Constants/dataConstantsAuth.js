export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
