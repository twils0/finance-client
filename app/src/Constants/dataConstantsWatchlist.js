export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
