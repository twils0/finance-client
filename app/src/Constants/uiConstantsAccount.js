export const actionTypes = {
  SET_CURRENT_FORM: 'ACCOUNT_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'ACCOUNT_SET_FORM_HEIGHT',
  SET_FORM_EDIT: 'ACCOUNT_SET_FORM_EDIT',
  SET_INPUT_VALUE_ERROR: 'ACCOUNT_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'ACCOUNT_SET_BUTTON_TEXT',
  SET_BUTTON_VISIBLE: 'ACCOUNT_SET_BUTTON_VISIBLE',
};

export const formNames = {
  CODE: 'codeForm',
  PROFILE: 'profileForm',
  BILLING: 'billingForm',
  CHANGE_PASSWORD: 'changePasswordForm',
  DELETE_ACCOUNT: 'deleteAccountForm',
};

export const buttonNames = {
  PROFILE: 'profileButton',
  BILLING: 'billingButton',
  CHANGE_PASSWORD: 'changePasswordButton',
  DELETE_ACCOUNT: 'deleteAccountButton',
  FIRST: 'firstButton',
  SECOND: 'secondButton',
};

export const buttonTexts = {
  RESEND: 'Resend',
  SUBMIT: 'Submit',
  EDIT: 'Edit',
  UPDATE: 'Update',
  CANCEL: 'Cancel',
  DELETE: 'Delete',
  CONFIRM: 'Confirm',
};

export const inputNames = {
  [formNames.CODE]: {
    CODE_PHONE: 'codeCodePhone',
  },
  [formNames.PROFILE]: {
    NAME: 'profileName',
    EMAIL: 'profileEmail',
    EMAIL_ADDITIONAL: 'profileEmailAdditional',
    PHONE: 'profilePhone',
  },
  [formNames.BILLING]: {
    NAME_ON_CARD: 'billingNameOnCard',
    PROMO_CODE: 'billingPromoCode',
    STRIPE: 'billingStripe',
  },
  [formNames.CHANGE_PASSWORD]: {
    OLD_PASSWORD: 'changePasswordOldPassword',
    NEW_PASSWORD: 'changePasswordNewPassword',
    NEW_PASSWORD2: 'changePasswordNewPassword2',
  },
  [formNames.DELETE_ACCOUNT]: {
    CONFIRM: 'deleteAccountConfirm',
  },
};
