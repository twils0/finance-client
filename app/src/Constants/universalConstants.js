import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
