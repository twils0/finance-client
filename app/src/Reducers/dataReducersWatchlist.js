import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, statusNames } from '../Constants/dataConstantsWatchlist';

export const initialState = {
  status: {
    [statusNames.SECURITIES]: {
      id: statusNames.SECURITIES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.GET_SECURITY_DATA]: {
      id: statusNames.GET_SECURITY_DATA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_SECURITIES]: {
      id: statusNames.UPDATE_SECURITIES,
      status: requestStatusTypes.IDLE,
    },
  },
  securities: {
    current: null,
    list: [],
  },
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_WATCHLIST_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_WATCHLIST_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const securities = (state = initialState.securities, action) => {
  switch (action.type) {
    case actionTypes.SET_SECURITIES_CURRENT:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_SECURITIES_LIST:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_SECURITIES_ALL:
      return {
        ...action.payload.securities,
      };
    case actionTypes.SET_SECURITY:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_SECURITY_DATA: {
      if (!state[action.payload.id]) {
        return {
          ...state,
          [action.payload.id]: {
            ...state[action.payload.id],
            ...action.payload,
          },
        };
      }

      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          data: {
            ...state[action.payload.id].data,
            ...action.payload.data,
          },
        },
      };
    }
    case actionTypes.RESET_WATCHLIST_STATE:
      return initialState.securities;
    default:
      return state;
  }
};

const dataReducersWatchlist = combineReducers({
  status,
  securities,
});

export default dataReducersWatchlist;
