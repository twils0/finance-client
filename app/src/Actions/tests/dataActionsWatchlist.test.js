import configureMockStore from 'redux-mock-store';
import { initialState } from '../../Reducers/dataReducersWatchlist';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { actionTypes } from '../../Constants/dataConstantsWatchlist';
import {
  setWatchlistStatus,
  setSecuritiesCurrent,
  setSecuritiesList,
  setSecuritiesAll,
  setSecurity,
  setSecurityData,
  resetWatchlistState,
} from '../dataActionsWatchlist';

const middleware = [];
const mockStore = configureMockStore(middleware);

describe('dataActionsWatchlist', () => {
  describe('setWatchlistStatus', () => {
    it("fails when not provided a payload with an 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { status: requestStatusTypes.SUCCESS };

      expect(() => store.dispatch(setWatchlistStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'status' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setWatchlistStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', status: requestStatusTypes.SUCCESS };
      const expectedActions = [
        {
          type: actionTypes.SET_WATCHLIST_STATUS,
          payload,
        },
      ];

      store.dispatch(setWatchlistStatus(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setSecuritiesCurrent', () => {
    it("fails when not provided a payload with a 'current' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setSecuritiesCurrent(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { current: 'testId' };
      const expectedActions = [
        {
          type: actionTypes.SET_SECURITIES_CURRENT,
          payload,
        },
      ];

      store.dispatch(setSecuritiesCurrent(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setSecuritiesList', () => {
    it("fails when not provided a payload with a 'list' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setSecuritiesList(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with a 'list' key that is not an array", () => {
      const store = mockStore(initialState);

      const payload = { list: {} };

      expect(() => store.dispatch(setSecuritiesList(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { list: ['testId1', 'testId2'] };
      const expectedActions = [
        {
          type: actionTypes.SET_SECURITIES_LIST,
          payload,
        },
      ];

      store.dispatch(setSecuritiesList(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setSecuritiesAll', () => {
    it("fails when not provided a payload with an 'securities' key", () => {
      const store = mockStore(initialState);

      const payload = {
        current: 'testId',
        list: ['testId1', 'testId2'],
        testId1: {
          id: 'testId1',
          name: 'testName1',
          exchange: null,
          tickerCusip: 'testTickerCusip1',
          category: 'testCategory1',
        },
        testId2: {
          id: 'testId2',
          name: 'testName2',
          exchange: null,
          tickerCusip: 'testTickerCusip2',
          category: 'testCategory2',
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an 'securities' key that is not an object", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: [],
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'current' key", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          list: ['testId1', 'testId2'],
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            id: 'testId2',
            name: 'testName2',
            exchange: null,
            tickerCusip: 'testTickerCusip2',
            category: 'testCategory2',
          },
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'list' key", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          current: 'testId1',
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            id: 'testId2',
            name: 'testName2',
            exchange: null,
            tickerCusip: 'testTickerCusip2',
            category: 'testCategory2',
          },
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with a 'list' key that is not an array", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          current: 'testId1',
          list: {},
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            id: 'testId2',
            name: 'testName2',
            exchange: null,
            tickerCusip: 'testTickerCusip2',
            category: 'testCategory2',
          },
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an security that does not have an 'id' key", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          current: 'testId1',
          list: ['testId1', 'testId2'],
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            name: 'testName2',
            exchange: null,
            tickerCusip: 'testTickerCusip2',
            category: 'testCategory2',
          },
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an security that does not have an 'name' key", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          current: 'testId1',
          list: ['testId1', 'testId2'],
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            id: 'testId2',
            exchange: null,
            tickerCusip: 'testTickerCusip2',
            category: 'testCategory2',
          },
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an security that does not have an 'exchange' key", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          current: 'testId1',
          list: ['testId1', 'testId2'],
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            id: 'testId2',
            name: 'testName2',
            tickerCusip: 'testTickerCusip2',
            category: 'testCategory2',
          },
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an security that does not have an 'tickerCusip' key", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          current: 'testId1',
          list: ['testId1', 'testId2'],
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            id: 'testId2',
            name: 'testName2',
            exchange: null,
            category: 'testCategory2',
          },
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an security that does not have an 'category' key", () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          current: 'testId1',
          list: ['testId1', 'testId2'],
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            id: 'testId2',
            name: 'testName2',
            exchange: null,
            tickerCusip: 'testTickerCusip2',
          },
        },
      };

      expect(() => store.dispatch(setSecuritiesAll(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = {
        securities: {
          current: 'testId1',
          list: ['testId1', 'testId2'],
          testId1: {
            id: 'testId1',
            name: 'testName1',
            exchange: null,
            tickerCusip: 'testTickerCusip1',
            category: 'testCategory1',
          },
          testId2: {
            id: 'testId2',
            name: 'testName2',
            exchange: null,
            tickerCusip: 'testTickerCusip2',
            category: 'testCategory2',
          },
        },
      };
      const expectedActions = [
        {
          type: actionTypes.SET_SECURITIES_ALL,
          payload,
        },
      ];

      store.dispatch(setSecuritiesAll(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setSecurity', () => {
    it("fails when not provided a payload with an 'id' key", () => {
      const store = mockStore(initialState);

      const payload = {
        name: 'testName',
        exchange: null,
        tickerCusip: 'testTickerCusip',
        category: 'testCategory',
      };

      expect(() => store.dispatch(setSecurity(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with an 'name' key", () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        exchange: null,
        tickerCusip: 'testTickerCusip',
        category: 'testCategory',
      };

      expect(() => store.dispatch(setSecurity(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with an 'exchange' key", () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        name: 'testName',
        tickerCusip: 'testTickerCusip',
        category: 'testCategory',
      };

      expect(() => store.dispatch(setSecurity(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with an 'tickerCusip' key", () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        name: 'testName',
        exchange: null,
        category: 'testCategory',
      };

      expect(() => store.dispatch(setSecurity(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with an 'category' key", () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        name: 'testName',
        exchange: null,
        tickerCusip: 'testTickerCusip',
      };

      expect(() => store.dispatch(setSecurity(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        name: 'testName',
        exchange: null,
        tickerCusip: 'testTickerCusip',
        category: 'testCategory',
      };
      const expectedActions = [
        {
          type: actionTypes.SET_SECURITY,
          payload,
        },
      ];

      store.dispatch(setSecurity(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setSecurityData', () => {
    it("fails when not provided a payload with an 'id' key", () => {
      const store = mockStore(initialState);

      const payload = {
        data: {
          description: 'testDescription',
        },
      };

      expect(() => store.dispatch(setSecurityData(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with an 'data' key", () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
      };

      expect(() => store.dispatch(setSecurityData(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with a 'data' key that is not an object", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', data: [] };

      expect(() => store.dispatch(setSecurityData(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = {
        id: 'testId',
        data: {
          description: 'testDescription',
        },
      };
      const expectedActions = [
        {
          type: actionTypes.SET_SECURITY_DATA,
          payload,
        },
      ];

      store.dispatch(setSecurityData(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('resetWatchlistState', () => {
    it('creates the correct action', () => {
      const store = mockStore(initialState);

      const expectedActions = [
        {
          type: actionTypes.RESET_WATCHLIST_STATE,
        },
      ];

      store.dispatch(resetWatchlistState());

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });
});
