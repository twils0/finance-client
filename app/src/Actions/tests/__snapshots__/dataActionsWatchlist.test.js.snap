// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataActionsWatchlist setSecuritiesAll fails when not provided a payload with a 'current' key 1`] = `"Please enter a value for the 'current' key - {\\"list\\":[\\"testId1\\",\\"testId2\\"],\\"testId1\\":{\\"id\\":\\"testId1\\",\\"name\\":\\"testName1\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip1\\",\\"category\\":\\"testCategory1\\"},\\"testId2\\":{\\"id\\":\\"testId2\\",\\"name\\":\\"testName2\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip2\\",\\"category\\":\\"testCategory2\\"}}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when not provided a payload with a 'list' key 1`] = `"Please enter a value for the 'list' key - {\\"current\\":\\"testId1\\",\\"testId1\\":{\\"id\\":\\"testId1\\",\\"name\\":\\"testName1\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip1\\",\\"category\\":\\"testCategory1\\"},\\"testId2\\":{\\"id\\":\\"testId2\\",\\"name\\":\\"testName2\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip2\\",\\"category\\":\\"testCategory2\\"}}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when not provided a payload with an 'securities' key 1`] = `"Please enter a value for the 'securities' key - {\\"current\\":\\"testId\\",\\"list\\":[\\"testId1\\",\\"testId2\\"],\\"testId1\\":{\\"id\\":\\"testId1\\",\\"name\\":\\"testName1\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip1\\",\\"category\\":\\"testCategory1\\"},\\"testId2\\":{\\"id\\":\\"testId2\\",\\"name\\":\\"testName2\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip2\\",\\"category\\":\\"testCategory2\\"}}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when provided a payload with a 'list' key that is not an array 1`] = `"Please enter an array value for the 'list' key - {\\"current\\":\\"testId1\\",\\"list\\":{},\\"testId1\\":{\\"id\\":\\"testId1\\",\\"name\\":\\"testName1\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip1\\",\\"category\\":\\"testCategory1\\"},\\"testId2\\":{\\"id\\":\\"testId2\\",\\"name\\":\\"testName2\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip2\\",\\"category\\":\\"testCategory2\\"}}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when provided a payload with an 'securities' key that is not an object 1`] = `"Please enter an object value for the 'securities' key - {\\"securities\\":[]}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when provided a payload with an security that does not have an 'category' key 1`] = `"Please enter a value for the 'category' key - {\\"id\\":\\"testId2\\",\\"name\\":\\"testName2\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip2\\"}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when provided a payload with an security that does not have an 'exchange' key 1`] = `"Please enter a value for the 'exchange' key - {\\"id\\":\\"testId2\\",\\"name\\":\\"testName2\\",\\"tickerCusip\\":\\"testTickerCusip2\\",\\"category\\":\\"testCategory2\\"}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when provided a payload with an security that does not have an 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"name\\":\\"testName2\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip2\\",\\"category\\":\\"testCategory2\\"}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when provided a payload with an security that does not have an 'name' key 1`] = `"Please enter a value for the 'name' key - {\\"id\\":\\"testId2\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip2\\",\\"category\\":\\"testCategory2\\"}"`;

exports[`dataActionsWatchlist setSecuritiesAll fails when provided a payload with an security that does not have an 'tickerCusip' key 1`] = `"Please enter a value for the 'tickerCusip' key - {\\"id\\":\\"testId2\\",\\"name\\":\\"testName2\\",\\"exchange\\":null,\\"category\\":\\"testCategory2\\"}"`;

exports[`dataActionsWatchlist setSecuritiesCurrent fails when not provided a payload with a 'current' key 1`] = `"Please enter a value for the 'current' key - {}"`;

exports[`dataActionsWatchlist setSecuritiesList fails when not provided a payload with a 'list' key 1`] = `"Please enter a value for the 'list' key - {}"`;

exports[`dataActionsWatchlist setSecuritiesList fails when provided a payload with a 'list' key that is not an array 1`] = `"Please enter an array value for the 'list' key - {\\"list\\":{}}"`;

exports[`dataActionsWatchlist setSecurity fails when not provided a payload with an 'category' key 1`] = `"Please enter a value for the 'category' key - {\\"id\\":\\"testId\\",\\"name\\":\\"testName\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip\\"}"`;

exports[`dataActionsWatchlist setSecurity fails when not provided a payload with an 'exchange' key 1`] = `"Please enter a value for the 'exchange' key - {\\"id\\":\\"testId\\",\\"name\\":\\"testName\\",\\"tickerCusip\\":\\"testTickerCusip\\",\\"category\\":\\"testCategory\\"}"`;

exports[`dataActionsWatchlist setSecurity fails when not provided a payload with an 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"name\\":\\"testName\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip\\",\\"category\\":\\"testCategory\\"}"`;

exports[`dataActionsWatchlist setSecurity fails when not provided a payload with an 'name' key 1`] = `"Please enter a value for the 'name' key - {\\"id\\":\\"testId\\",\\"exchange\\":null,\\"tickerCusip\\":\\"testTickerCusip\\",\\"category\\":\\"testCategory\\"}"`;

exports[`dataActionsWatchlist setSecurity fails when not provided a payload with an 'tickerCusip' key 1`] = `"Please enter a value for the 'tickerCusip' key - {\\"id\\":\\"testId\\",\\"name\\":\\"testName\\",\\"exchange\\":null,\\"category\\":\\"testCategory\\"}"`;

exports[`dataActionsWatchlist setSecurityData fails when not provided a payload with an 'data' key 1`] = `"Please enter a value for the 'data' key - {\\"id\\":\\"testId\\"}"`;

exports[`dataActionsWatchlist setSecurityData fails when not provided a payload with an 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"data\\":{\\"description\\":\\"testDescription\\"}}"`;

exports[`dataActionsWatchlist setSecurityData fails when provided a payload with a 'data' key that is not an object 1`] = `"Please enter an object value for the 'data' key - {\\"id\\":\\"testId\\",\\"data\\":[]}"`;

exports[`dataActionsWatchlist setWatchlistStatus fails when not provided a payload with a 'status' key 1`] = `"Please enter a value for the 'status' key - {\\"id\\":\\"testId\\"}"`;

exports[`dataActionsWatchlist setWatchlistStatus fails when not provided a payload with an 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"status\\":\\"SUCCESS\\"}"`;
